### PR TITLE
Update backup file when deleting a snapshot

### PR DESCRIPTION
**Pull request by feileacan**
Fixes https://github.com/lxc/lxd/issues/8227

Signed-off-by: David Munro <davidmunro@google.com>

### CODE REVIEW

**Comment by feileacan** [comment]
Still todo when I have time: test and (I guess?) qemu driver.

File: lxd/instance/drivers/driver_lxc.go [hunk 1]
```diff
@@ -3397,6 +3397,10 @@ func (d *lxc) Delete(force bool) error {
 				if err != nil {
 					return err
 				}
+				err = c.UpdateBackupFile()
```
Comment (tomponline) [inline lxd/instance/drivers/driver_lxc.go:3400]: @feileacan please can you move this down 1 line so there is a gap from the previous if statement. Thanks

**Comment by tomponline** [comment]
@feileacan thanks for this, yes, as you say, qemu will need it as well.

**Comment by stgraber** [comment]
Looking into this branch now as we have a release coming up soon and would like this sorted then :)

**Comment by tomponline** [comment]
@stgraber something I was intending to look at also is that there is some inconistency between the calls to backup the file between lxc and qemu that probably need aligning or at least commenting to explain difference. Lxc has more calls to it than qemu for instance.

**Comment by stgraber** [comment]
@tomponline added a commit to fix that

**Comment by stgraber** [comment]
jenkins: test this please

**Comment by stgraber** [comment]
@feileacan so FYI, your approach wouldn't work as that function can only be called on an instance, not on a snapshot, it's just a no-op when called on a snapshot.

I've also move the logic down to when the database record of the snapshot is fully gone just to avoid potential issues with the database.

**Review by tomponline — APPROVED** [review]

**Comment by stgraber** [comment]
@tomponline good to go

**Pull request merged**